File: MPChartLib/src/main/java/com/github/mikephil/charting/interfaces/datasets/IScatterDataSet.java
package com.github.mikephil.charting.interfaces.datasets;

import com.github.mikephil.charting.data.Entry;
import com.github.mikephil.charting.renderer.scatter.IShapeRenderer;

public interface IScatterDataSet extends ILineScatterCandleRadarDataSet<Entry> {

    /**
     * the currently set scatter shape size
     */
    float getScatterShapeSize();

    /**
     * radius of the hole in the shape
     */
    float getScatterShapeHoleRadius();

    /**
     * the color for the hole in the shape
     */
    int getScatterShapeHoleColor();

    /**
     * the IShapeRenderer responsible for rendering this DataSet.
     */
    IShapeRenderer getShapeRenderer();
}

File: MPChartLib/src/main/java/com/github/mikephil/charting/interfaces/datasets/IScatterDataSet.kt
package com.github.mikephil.charting.interfaces.datasets

import com.github.mikephil.charting.data.Entry
import com.github.mikephil.charting.renderer.scatter.IShapeRenderer

interface IScatterDataSet : ILineScatterCandleRadarDataSet<Entry> {
    /**
     * the currently set scatter shape size
     */
    val scatterShapeSize: Float

    /**
     * radius of the hole in the shape
     */
    val scatterShapeHoleRadius: Float

    /**
     * the color for the hole in the shape
     */
    val scatterShapeHoleColor: Int

    /**
     * the IShapeRenderer responsible for rendering this DataSet.
     */
    val shapeRenderer: IShapeRenderer?
}
